clients/java/signalr/src/test/java/HubConnectionTest.java

```diff
@@ -393,4 +419,35 @@ public void onReceive(String message) throws Exception {
         @Override
         public void stop() {return;}
     }
+
+    private class HandshakeTransport implements Transport {
```

Comment:
I think it would be cleaner and more reusable to just build a proper `MockTransport` like we have in C# (in C# it's `TestConnection` that does this work). It would store a list of all the messages that have been sent so far and provide a method that could trigger `onReceive`. So a test would look something like:

```java
MockTransport transport = new MockTransport();
HubConnection hubConnection = new HubConnection("http://example.com", transport);

hubConnection.on("inc", () ->{
    assertEquals(0.0, value.get(), 0);
    value.getAndUpdate((val) -> val + 1);
});

hubConnection.start();

assertEqual(transport.getSentMessages()[0], "{... handshake request ...}");

transport.receiveMessage("{ ... handshake response ... }" + RECORD_SEPARATOR + "{ ... invocation ... }" + RECORD_SEPARATOR);
```

This `HandshakeTransport` is working fine, but seems a little funky, with the message in `send` being ignored, and it's kinda just a single-purpose component.

Reply:
Yeah, the public `receiveMessage ` on the transport is what I need 😄

Reply:
That cleans up some of the trickery in the test transports

Reply:
I'll add the method in the MockTransport and change the new tests to use it for now. I'll update the rest of the tests in a later PR.